DopplerBillingJob/Database/DopplerRepository.cs

```diff
@@ -4,40 +4,34 @@
 using System.Threading.Tasks;
 using CrossCutting.DopplerSapService.Entities;
 using Dapper;
-using Doppler.Billing.Job.Settings;
 using Doppler.Database;
 using Microsoft.Extensions.Logging;
-using Microsoft.Extensions.Options;
 
 namespace Doppler.Billing.Job.Database
 {
     public class DopplerRepository : IDopplerRepository
     {
         private readonly ILogger<DopplerRepository> _logger;
-        private readonly IOptionsMonitor<DopplerBillingJobSettings> _dopplerBillingJobSettings;
         private readonly IDbConnectionFactory _dbConnectionFactory;
 
         public DopplerRepository(
             ILogger<DopplerRepository> dopplerRepositoryLogger,
-            IOptionsMonitor<DopplerBillingJobSettings> dopplerBillingJobSettings,
             IDbConnectionFactory dbConnectionFactory)
         {
             _logger = dopplerRepositoryLogger;
-            _dopplerBillingJobSettings = dopplerBillingJobSettings;
             _dbConnectionFactory = dbConnectionFactory;
         }
 
-        public async Task<IList<UserBilling>> GetUserBillingInformation()
+        public async Task<IList<UserBilling>> GetUserBillingInformation(List<string> storedProcedures)
```

Comment:
Why did you refactor it moving an object level injection to a method level injection?

Reply:
My idea was that the each billing job gets the stored procedures that it need to runs and send them to the repository and remove the logic  to determinate witch stored procedures need to runs depending the job in the repository. What do you think?

Reply:
So, in the future, you are planning to have different implementations of the current `DopplerBillingJob` and only one `IDopplerRepository`?

Reply:
Currently we have 2 Billing jobs : ```DopplerBillingJob``` and ```DopplerBillingUsJob``` and both use the same Repository. Each job pass the corresponding stored procedures:

**DopplerBillingJob:** (SAP AR)

```var billingData = await _dopplerRepository.GetUserBillingInformation(_dopplerBillingJobSettings.CurrentValue.StoredProcedures);```

**DopplerBillingUsJob:** (SAP US)

```var billingData = await _dopplerRepository.GetUserBillingInformation(_dopplerBillingUsJobSettings.CurrentValue.StoredProcedures);```

Let me know if you want a meeting to explain you with more details.

Reply:
No, I think that I am understanding, it makes sense.